src/scripts/main.js

```diff
@@ -1,3 +1,329 @@
+const button = document.querySelector('.start');
+const messageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+const counter = document.querySelector('.game-score');
+
+const rows = document.querySelectorAll('tr');
+const cells = document.querySelectorAll('td');
+
+const tableSize = 4;
+
+const arrayCells = new Array(tableSize);
+
+for (let i = 0; i < arrayCells.length; i++) {
+  arrayCells[i] = new Array(tableSize).fill(0);
+}
+
+let handleKeyDown = false;
+
+button.addEventListener('click', e => {
+  handleKeyDown = true;
+
+  if (!button.classList.contains('restart')) {
+    button.textContent = 'Restart';
+    button.classList.add('restart');
+    button.classList.remove('start');
+    messageStart.classList.add('hidden');
+  } else {
+    for (const element of arrayCells) {
+      element.fill(0);
+    }
+
+    updateTable();
+  }
+
+  generateRandomCell();
+  generateRandomCell();
+});
+
+function checkFreeSpace() {
+  for (const row of arrayCells) {
+    if (row.includes(0)) {
+      return true;
+    }
+  }
+
+  return false;
+}
+
+function generateRandomCell() {
+  if (!checkFreeSpace()) {
+    return;
+  }
+
+  let rowIndex, cellIndex;
+
+  do {
+    rowIndex = Math.floor(Math.random() * tableSize);
+    cellIndex = Math.floor(Math.random() * tableSize);
+  } while (arrayCells[rowIndex][cellIndex] !== 0);
+
+  if (arrayCells[rowIndex][cellIndex] === 0) {
+    const newValue = Math.random() < 0.75 ? 2 : 4;
+
+    arrayCells[rowIndex][cellIndex] = newValue;
+
+    const cell = rows[rowIndex].children[cellIndex];
+
+    cell.textContent = newValue;
+    cell.classList.add(`field-cell--${newValue}`);
+  }
+}
+
+document.addEventListener('keydown', e => {
+  if (handleKeyDown) {
+    switch (e.key) {
+      case 'ArrowRight':
+        moveRight();
+        generateRandomCell();
+
+        if (checkGameOver()) {
+          messageLose.classList.remove('hidden');
+        }
+
+        if (checkWin()) {
+          messageWin.classList.remove('hidden');
+        }
+        break;
+
+      case 'ArrowLeft':
+        moveLeft();
+        generateRandomCell();
+
+        if (checkGameOver()) {
+          messageLose.classList.remove('hidden');
+        }
+
+        if (checkWin()) {
+          messageWin.classList.remove('hidden');
+        }
+        break;
+
+      case 'ArrowUp':
+        moveUp();
+        generateRandomCell();
+
+        if (checkGameOver()) {
+          messageLose.classList.remove('hidden');
+        }
+
+        if (checkWin()) {
+          messageWin.classList.remove('hidden');
+        }
+        break;
+
+      case 'ArrowDown':
+        moveDown();
+        generateRandomCell();
+
+        if (checkGameOver()) {
+          messageLose.classList.remove('hidden');
+        }
+
+        if (checkWin()) {
+          messageWin.classList.remove('hidden');
+        }
+        break;
+
+      default:
```

Comment:
Consider removing the `throw new Error` statement, since the game should not throw an error if a non- arrow key is pressed. Instead, do nothing or log a message to the console. 🎮